writeengine/xml/we_xmljob.cpp

```diff
@@ -1087,13 +1087,24 @@ void XMLJob::fillInXMLDataNotNullDefault(
             case execplan::CalpontSystemCatalog::DECIMAL:
             case execplan::CalpontSystemCatalog::UDECIMAL:
             {
-                col.fDefaultInt = Convertor::convertDecimalString(
-                                      col_defaultValue.c_str(),
-                                      col_defaultValue.length(),
-                                      colType.scale);
+                if (LIKELY(colType.colWidth == datatypes::MAXDECIMALWIDTH))
+                {
+                    bool dummy = false;
+                    dataconvert::number_int_value(col_defaultValue, colType.colDataType,
```

Comment:
I would like to discourage you from using dataconvert functions and use TypeHandler classes' methods instead. Could you leverage SystemCatalog::TypeAttributesStd::decimal128FromString() here?

Reply:
Thanks for pointing this out, I wasn't aware of this function. Patch is updated to use `decimal128FromString` here.